controller/registries/consul/config.go

```diff
@@ -102,15 +121,20 @@ func (reg *Consul) NewClient(config *consul.Config) (*Client, error) {
 func (reg *Consul) Start(c registrytype.RegistryConfig) error {
 	config := c.(*consul.Config)
 
-	client, err := reg.NewClient(config)
-	if err != nil {
-		return err
-	}
+	if reg.client == nil {
```

Comment:
So far the `Start` would be called only one. We can skip the `reg.client == nil` check?